torchcam/utils.py

```diff
@@ -42,6 +42,10 @@ def overlay_mask(img: Image, mask: Image, colormap: str = "jet", alpha: float =
     cmap = cm.get_cmap(colormap)
     # Resize mask and apply colormap
     overlay = mask.resize(img.size, resample=Resampling.BICUBIC)
-    overlay = (255 * cmap(np.asarray(overlay) ** 2)[:, :, :3]).astype(np.uint8)
+    if len(img.getbands()) == 1:
+        overlay = (255 * cmap(np.asarray(overlay) ** 2)[:, :, 0]).astype(np.uint8)
+    else:
+        overlay = (255 * cmap(np.asarray(overlay) ** 2)[:, :, :3]).astype(np.uint8)
+
```

Comment:
Let's make this even more robust: 
- the visualization can only work properly for single channel & 3 channel colormaps
- so let's raise an AssertionError if we don't have one of those two even before creating the `cmap`.
- since we're only doing broadcasting, I suggest to do:
```python
overlay = (255 * cmap(np.asarray(overlay) ** 2)[:, :, :len(img.getbands())]).astype(np.uint8)
```

Reply:
```
overlay = (255 * cmap(np.asarray(overlay) ** 2)[:, :, 2 if len(img.getbands())==1 else slice(0, 3)]).astype(np.uint8)
```

will be more correct